engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/RequestResponseAkkaHttpHandler.scala

```diff
@@ -0,0 +1,98 @@
+package pl.touk.nussknacker.engine.embedded
+
+import akka.event.Logging
+import akka.http.scaladsl.model.MediaTypes.`application/json`
+import akka.http.scaladsl.model.{HttpEntity, HttpResponse, ResponseEntity, StatusCodes}
+import akka.http.scaladsl.server.{Directive0, Directive1, Directives, Route}
+import akka.http.scaladsl.server.directives.DebuggingDirectives
+import akka.stream.Materializer
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.{Encoder, Json}
+import io.circe.generic.JsonCodec
+import io.circe.syntax.EncoderOps
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
+import pl.touk.nussknacker.engine.requestresponse.DefaultResponseEncoder
+import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+import pl.touk.nussknacker.engine.requestresponse.RequestResponseEngine.RequestResponseResultType
+import pl.touk.nussknacker.engine.requestresponse.api.{RequestResponseGetSource, RequestResponsePostSource}
+import pl.touk.nussknacker.engine.requestresponse.metrics.InvocationMetrics
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class ProcessRoute(processInterpreters: scala.collection.Map[String, InterpreterType]) extends Directives with LazyLogging {
+
+  protected def logDirective(scenarioName: String): Directive0 = DebuggingDirectives.logRequestResult((s"request-response-$scenarioName", Logging.DebugLevel))
+
+  def route(implicit ec: ExecutionContext, mat: Materializer): Route =
+    path(Segment) { processPath =>
+      logDirective(processPath) {
+        processInterpreters.get(processPath) match {
+          case None =>
+            complete {
+              HttpResponse(status = StatusCodes.NotFound)
+            }
+          case Some(processInterpreter) => new RequestResponseAkkaHttpHandler(processInterpreter).invoke {
+            case Invalid(errors) => complete {
+              logErrors(processPath, errors)
+              HttpResponse(status = StatusCodes.InternalServerError, entity = toEntity(errors.toList.map(info => EspError(info.nodeComponentInfo.map(_.nodeId), Option(info.throwable.getMessage)))))
+            }
+            case Valid(results) => complete {
+              HttpResponse(status = StatusCodes.OK, entity = toEntity(results))
+            }
+          }
+        }
+      }
+      //TODO place openApi endpoint
+    } ~ pathEndOrSingleSlash {
```

Comment:
can it be explicit named like /ready or sth? or u want to be backward compatible?

Reply:
1. I just copied from current usage
2. I wanted to avoid edge cases like "what if I configure scenario's path to 'ready' etc. ;)

Reply:
1. Is it important? I thought that we are not thinking about backward compatibility in this place. Do someone use this api on production now?
2. So maybe some prefix for scenario requests like /scenario and /ready on the same level? Handling slash may introduce bugs. What about situation when in case of some bug like empty scenario name, someone will invoke slash? It will got 200 OK and will think that everything is fine. This case should fail fast.

Reply:
sounds good, I changed to /scenario and /healthCheck (to be consistent with the designer)